native/src/routes/jpal-tracking/components/__tests__/JpalTracking.spec.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import React from 'react'
+import JpalTracking from '../JpalTracking'
+import lightTheme from '../../../../modules/theme/constants'
+import { fireEvent, render } from '@testing-library/react-native'
+import createNavigationMock from '../../../../testing/createNavigationPropMock'
+import AsyncStorage from '@react-native-community/async-storage'
+import type { RoutePropType } from '../../../../modules/app/constants/NavigationTypes'
+import type { JpalTrackingRouteType } from 'api-client'
+import { JPAL_TRACKING_ROUTE } from 'api-client'
+
+jest.mock('@react-native-community/async-storage')
+
+describe('JpalTracking', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks()
+    await AsyncStorage.clear()
+  })
+  const t = key => key
+  const navigation = createNavigationMock()
+  const route: RoutePropType<JpalTrackingRouteType> = {
+    key: 'route-id-0',
+    params: { trackingCode: 'abc' },
+    name: JPAL_TRACKING_ROUTE
+  }
+
+  it('Textfield should not be editable, when tracking is disabled', async () => {
+    const { findByTestId } = render(<JpalTracking theme={lightTheme} t={t} route={route} navigation={navigation} />)
+
+    AsyncStorage.setItem('jpalTrackingEnabled', true, async () => {
```

Comment:
I'd rather use our AppSettings instead of AsyncStorage directly. I also think it is weird to render the component before you set the setting initially. Maybe it also makes sense to assert on a few more things like that jpal tracking is enabled in the AppSettings if you toggled the switch.